CondFormats/RPCObjects/test/RPCReadOutMapAnalyzer.cc

```diff
@@ -17,38 +17,34 @@ using namespace std;
 using namespace edm;
 
 // class declaration
-class RPCReadOutMapAnalyzer : public edm::EDAnalyzer {
+class RPCReadOutMapAnalyzer : public edm::one::EDAnalyzer<> {
 public:
   explicit RPCReadOutMapAnalyzer(const edm::ParameterSet&);
-  ~RPCReadOutMapAnalyzer();
-  virtual void analyze(const edm::Event&, const edm::EventSetup&);
+  ~RPCReadOutMapAnalyzer() override = default;
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
 
 private:
   bool m_flag;
+  edm::ESGetToken<RPCEMap, RPCEMapRcd> readoutMappingToken_;
+  edm::ESGetToken<RPCReadOutMapping, RPCReadOutMappingRcd> mapZeroToken_;
 };
 
 RPCReadOutMapAnalyzer::RPCReadOutMapAnalyzer(const edm::ParameterSet& iConfig)
-    : m_flag(iConfig.getUntrackedParameter<bool>("useNewEMap", false)) {
+    : m_flag(iConfig.getUntrackedParameter<bool>("useNewEMap", false)),
+      readoutMappingToken_(esConsumes()),
+      mapZeroToken_(esConsumes()) {
   ::putenv(const_cast<char*>(std::string("CORAL_AUTH_USER konec").c_str()));
   ::putenv(const_cast<char*>(std::string("CORAL_AUTH_PASSWORD konecPass").c_str()));
 }
 
-RPCReadOutMapAnalyzer::~RPCReadOutMapAnalyzer() {}
-
 void RPCReadOutMapAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
   std::cout << "====== RPCReadOutMapAnalyzer" << std::endl;
 
   const RPCReadOutMapping* map;
   if (m_flag) {
-    edm::ESHandle<RPCEMap> readoutMapping;
-    iSetup.get<RPCEMapRcd>().get(readoutMapping);
-    const RPCEMap* eMap = readoutMapping.product();
-    //     RPCReadOutMapping* map = eMap->convert();
-    map = eMap->convert();
+    map = iSetup.getHandle(readoutMappingToken_).product()->convert();
   } else {
-    edm::ESHandle<RPCReadOutMapping> map0;
-    iSetup.get<RPCReadOutMappingRcd>().get(map0);
-    map = map0.product();
+    map = iSetup.getHandle(mapZeroToken_).product();
```

Comment:
Hi @mrodozov this is a bit different from what I suggested. What suggested had the `getData()` function.
`getData()` is like the `getHandle` and the `.product()` together

I also suggested to make it `const` (and to use `auto`)

Reply:
```suggestion
    const auto map = &iSetup.getData(mapZeroToken_);
```

Reply:
the map is defined outside the condition already